libyul/optimiser/NameSimplifier.cpp

```diff
@@ -109,8 +110,10 @@ void NameSimplifier::findSimplification(YulString _name)
 	}
 	if (name != _name.str())
 	{
-		m_usedNames.insert(YulString(name));
-		m_translations[_name] = YulString(name);
+		YulString newName{name};
+		m_usedNames.insert(newName);
```

Comment:
Do we actually still need m_usedNames if the dispenser keeps track?

Also, in general: Does the dispenser mark names as available again if they are deleted?

Reply:
> Do we actually still need m_usedNames if the dispenser keeps track?

Let me check.

> Also, in general: Does the dispenser mark names as available again if they are deleted?

No we don't do that. I think it's better if we don't do this.

Reply:
At least at the end we should do it. If we change the name simplifier to run once at the end (in addition to once at the beginning) it should be able to re-use names that were used once during the optimization process but are not used at the end.